### PR TITLE
Reimplement lamp using wot-td

### PR DESCRIPTION
**Pull request by dodomorandi**
A rough reimplementation of lamp using wot-td. It barely passes the tests in webthing-tester.

We probably need to discuss whether we need to implement a slightly bigger/better set of features.

- Depends on https://github.com/sifis-home/wot-td/pull/11

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,6 +8,17 @@ edition = "2021"
 [dependencies]
 webthing = "0.15.0"
```
Comment (lu-zero) [inline Cargo.toml:9]: shouldn't it be removed at the same time?

File: src/bin/lamp.rs [hunk 2]
```diff
@@ -1,228 +1,782 @@
...
+        Action::Fade(fade) => {
+            let action = state.fade(fade.input).await;
+            let (input, href) = match action.ty {
+                StoredActionType::Fade(StoredFadeAction { input, href, .. }) => (input, href),
+            };
+
+            let response = (
+                StatusCode::CREATED,
+                Json(json!({
+                    "fade": {
```
Comment (lu-zero) [inline src/bin/lamp.rs:684]: It should be an [ActionStatus](https://w3c.github.io/wot-profile/#ActionStatus) and not start with the named wrapper.

File: Cargo.toml (hunk 1 above)
Comment (dodomorandi) [reply Cargo.toml:9]: Yes, I totally did not update Cargo.toml. Probably there are other things that can be safely removed as well.

File: src/bin/lamp.rs (hunk 2 above)
Comment (dodomorandi) [reply src/bin/lamp.rs:684]: I think you are almost surely right... which is quite bad, because it demonstrates that the tests are pretty unreliable...

**Comment by lu-zero** [comment]
We could land this and then work on getting the tests right.

**Comment by lu-zero** [comment]
This PR a candidate to measure the before-after regarding our quality metrics @Luni-4

File: src/bin/lamp.rs [hunk 3]
```diff
@@ -1,228 +1,647 @@
...
+    Json(value): Json<Value>,
+    Extension(state): Extension<AppState>,
+) -> Result<impl IntoResponse, AppError> {
+    match action_name {
+        ActionName::Fade => {
+            let fade: FadeActionInput =
+                serde_json::from_value(value).map_err(|_| AppError::InvalidValue)?;
+
+            let action = state.fade(fade).await;
+            let (output, href) = match action.ty {
```
[the remark this answers is not in the document]
Comment (dodomorandi) [reply src/bin/lamp.rs:630]: You are totally right, there are some places in which I use a `match` with just one variant, and the reason is I expected more than one action/property/whatever. In the end, I left the `match`es, and it is better to convert them to `let if`.

[the remark this answers is not in the document]
Comment (dodomorandi) [reply src/bin/lamp.rs]: I refactored some bits, just to move the _dummy matches_ in dedicated functions. In the end I left the `match`es, using a `let if` (or even a `let else` in the future) would only make things more verbose (in a "real" code, it does not make any sense to have a one variant enum :sweat_smile:). Let me know if now it is better for you :relaxed:

**Pull request merged**